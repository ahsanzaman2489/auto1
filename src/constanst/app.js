export const DEFAULT_PAGE_LIMIT = 10;
export const HOST = "http://localhost:";
export const HOST_PORT = 3001;
export const HEADERS = {"Access-Control-Allow-Origin":"*"};