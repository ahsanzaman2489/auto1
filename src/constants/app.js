export const HOST = "http://localhost:";
export const HOST_PORT = 3001;
export const NO_DATA = 'Opps...! No results found with this query';
export const NO_FAVOURITE = 'If you like this car, click the button and save it in your collection of favourite items.';
export const FAVOURITE = 'This Car is your favourite';
export const FOOTER_TEXT = `${new Date().getFullYear()} Â© AUTO1 Group`;