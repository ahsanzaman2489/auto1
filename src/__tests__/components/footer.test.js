import React from 'react';
import FooterComponent from '../../components/footer';
import {FOOTER_TEXT} from '../../constants/app';

import {shallow} from 'enzyme';
import renderer from "react-test-renderer";

const setup = (props = {}, state = null) => {
    const defaultProps = {};
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<FooterComponent {...setUpProps}/>);
    const tree = () => renderer.create(<FooterComponent {...setUpProps}/>).toJSON();
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps, tree}
};

describe("Favourite component", () => {
    it('should match with snapshot', () => {
        const {tree} = setup();
        expect(tree()).toMatchSnapshot();
    });
    it('Should render correct text of footer', () => {
        const {wrapper} = setup();
        expect(wrapper.find("footer").text()).toBe(FOOTER_TEXT);
    });

});


