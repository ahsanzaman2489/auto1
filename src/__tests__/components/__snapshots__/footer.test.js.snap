// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourite component should match with snapshot 1`] = `
<footer>
  2019 Â© AUTO1 Group
</footer>
`;
